src/app/joke.service.ts

```diff
@@ -0,0 +1,56 @@
+// joke.service.ts
+
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class JokeService {
+  jokeResult: string; // stores value of random joke
+  favs = []; // arr that stores list of favorite jokes
+  searchList: any = []; // arr that stores list of search results
+  catList: any = [];
+  jokeUrl = 'https://api.chucknorris.io/jokes'; // url to get random joke
+  searchUrl = 'https://api.chucknorris.io/jokes/search?query='; // url to search for jokes; requires input
+  categoryUrl = 'https://api.chucknorris.io/jokes/categories'; // url to retrieve list of categories
+  searchTerm: string; // input for search
+
+  constructor(private http: HttpClient) { }
+
+  // retrieve and display joke
+  getJoke() {
+    interface JokeTime { // work with data retrieved
```

Comment:
Interfaces are usually imported from other files. Try making a file for specifically defining interfaces. This allows for them to be reused/accessed easier elsewhere.